puppetry/kinectv2/pygame_button.py

```diff
@@ -0,0 +1,167 @@
+
+#Pygame Colors
+# pgBlack,pgWhite,pgDarkGrey,pgLightGrey = (0,0,0),(255,255,255),(150,150,150),(211,211,211)
+# pgRed,pgDarkRed,pgLightRed,pgReallyLightRed = (255,0,0),(150,0,0),(255,100,100),(255,200,200)
+# pgGreen,pgDarkGreen,pgLightGreen,pgReallyLightGreen = (0,255,0),(0,150,0),(100,255,100),(200,255,200)
+# pgBlue,pgDarkBlue,pgLightBlue,pgReallyLightBlue = (0,0,255),(0,0,150),(100,100,255),(200,200,255)
+# pgYellow,pgDarkYellow,pgGold = (255,255,0),(200,200,0),(212,175,55)
+
+
+#Display images
+# def imageDisplay(Image_name,Width,Height,x,y):
+#     Image = pygame.image.load(f"{Image_name}")
+#     Image = pygame.transform.scale(Image,(Width,Height))
+#     ImageRect = Image.get_rect()
+#     ImageRect.center = (x,y)
+#     Window.blit(Image,ImageRect)
+#     return ImageRect.center,Image
+
+# For logging, set log_function to a function that expects a string
+import puppetry
+
+PYGLogger = puppetry.log
+
+def log_msg(message):
+    """ Send out string to a log """
+    if PYGLogger is not None:
+        PYGLogger(message)
+
+
+
+
+# Simple button class for pygame_x
+class PYGButton(object):
+
+    def __init__(self, center_x, center_y, width, height, 
+                button_color, hover_color,
+                name, text, text_size, text_color, click_callback = None):
+        """ make a button """
+        self._left = center_x - (width // 2)
+        self._top = center_y - (height // 2)
+        self._width = width
+        self._height = height
+        self._right = self._left + width
+        self._bottom = self._top + height
+        self._color = button_color
+        self._hover_color = hover_color
+        self._name = name
+        self._label = text
+        self._text_size = text_size
+        self._text_color = text_color
+        self._click_callback = click_callback
+        self._mouse_down = False
+
+        self._font = pygame.font.Font("freesansbold.ttf", text_size)
+
+        font_size = self._font.size(self._label)
+        self._text_x = (center_x - (font_size[0] // 1.5))
+        self._text_y = (center_y - (font_size[1] // 2))
+
+
+    def drawNormal(self, surface):
+        """ Draw normal frame, color and text for button """
+        pygame.draw.rect(surface, self._color, (self._left, self._top, self._width, self._height))              
+        rendered_text = self._font.render(F"{self._label}",True, self._text_color)
+        surface.blit(rendered_text, [self._text_x, self._text_y])
+
+
+    def drawHover(self, surface, mouse_down):
+        """ Draw hover frame, color and text for button  TO DO - use mouse_down, add _text_hover_color ?"""
+        pygame.draw.rect(surface, self._hover_color, (self._left, self._top, self._width, self._height))              
+        rendered_text = self._font.render(F"{self._label}",True, self._text_color)
+        shift = 4 if mouse_down else 0  
+        surface.blit(rendered_text, [self._text_x + shift, self._text_y + shift])
+
+
+    def mouseOverButton(self, mouse_pos):

```

Comment:
Method name violates PEP-8, etc... will not mention the rest but you get the point.